karelics_vesc_can_driver/battery_status.py

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/env python3
+from functools import partial
+import re
+import sys
+
+import rclpy
+from rclpy.node import Node, NodeNameNonExistentError
+from sensor_msgs.msg import BatteryState
+
+from karelics_vesc_can_driver.msg import VescStatus
+
+
+class BatteryStatus(Node):
+    def __init__(self):
+        super().__init__("battery_status")
+
+        self.battery_pub = self.create_publisher(BatteryState, "/battery", qos_profile=1)
+        self.battery_pub_timer = self.create_timer(1.0, self.publish_battery_percentage)
+
+        self.vesc_status_subscribers = dict()
+        self.vesc_voltages = dict()
+
+        self.current_battery_voltage = None
+
+        # voltage, percentage. All the other voltage values between these key-values
+        # are linearly calculated
+        self.key_voltages = [[54.10, 1.0],
+                             [48, 0.6],
+                             [45.8, 0.2],
+                             [44.2, 0.05],
+                             [41.8, 0.02],
+                             [40, 0]]
```

Comment:
This is never changed right, so it would be more appropriate to make it a class variable or a module constant. To make it even more explicit that they should not change you can make them a tuple (That is immutable)

Reply:
Yes these never change! Thanks for the tips, will make them into a class variable and a tuple so that it is immutable.